test/Chainweb/Test/Pact/TTL.hs

```diff
@@ -360,22 +357,24 @@ assertDoPreBlockFailure action = try @_ @PactException action >>= \case
 data Ctx = Ctx
     { _ctxMempool :: !(MVar MemPoolAccess)
     , _ctxQueue :: !PactQueue
-    , _ctxPdb :: !(PayloadDb HashMapCas)
```

Comment:
I have to ask, what is the justification for `PayloadDb` exposing the type of it's underlying store? `Foo cas` and associated constraints are all over PactService etc. It would be nice if downstream users didn't have to have these constraints as they don't give any value to clients; this is an example of that as a unit test shouldn't have to care (or employ existentials itself to hide it).

Reply:
The reason was to be able to use different storage backends. Originally, in the first versions of chainweb, we didn't use a persisted payload store at all. Exposing the type parameter allowed to swap in the persisted store once we had it ready.

Reply:
An obvious solution would be to wrap it in a non-parameterized data type and quantify over the type parameter, but that would result in worse performance.